models/users.js

```diff
@@ -0,0 +1,166 @@
+'use strict';
+
+var Db        = require('/../config/database');
+var crypto    = require('crypto');
+var validator = require('validator');
+
+var UserSchema = mongoose.Schema({
+    username : { type: String, required: true },
+    email    : { type: String, required: true },
+    password : { type: String, required: true },
+    token    : { type: String, required: true }
+});
+
+var UserMongoModel = Db.model('users', UserSchema);
+
+// Generates a random unique token ID
+function generateToken() {
+    var token;
+
+    crypto.randomBytes(256, function(err, nonce) {
+        // TODO: Handle this error
+        token = nonce.toString('base64');
+    });
+    return token;
+}
```

Comment:
Javascript is function-scoped, which is a lot different than block-scope, which is what you're probably familiar with.

Here, the function will always return undefined because the `return` statement is called before the `crypto.randomBytes` callback can finish.

I suggest you look more up on Javascript's scope, because it's really not intutitive